lib/Runtime/Library/JavascriptObject.cpp

```diff
@@ -1096,6 +1096,92 @@ namespace Js
         return JavascriptOperators::GetOwnEnumerablePropertyNames(object, scriptContext);
     }
 
+    Var JavascriptObject::GetValuesOrEntries(RecyclableObject* object, bool valuesToReturn, ScriptContext* scriptContext)
+    {
+        Assert(object != nullptr);
+        Assert(scriptContext != nullptr);
+        JavascriptArray* valuesArray = scriptContext->GetLibrary()->CreateArray(0);
+
+        Var ownKeysVar = JavascriptOperators::GetOwnEnumerablePropertyNames(object, scriptContext);
```

Comment:
Good catch - the [spec](https://github.com/tc39/proposal-object-values-entries/blob/master/spec.md#enumerableownproperties-abstract-operation) says that the list of existing own keys is fixed at the initial call, but that means a preexisting own key's enumerability can change in either direction during enumeration.